src/Python/logger.py

```diff
@@ -1,37 +1,68 @@
 # Read and Store MPU6050 Data to Disk
-
+import csv
 import serial
 from serial.serialutil import SerialException
 
 import time
 import argparse
+import os
+
+# Create timestamp to log entries and create log file
+date_str = time.strftime("%Y-%m-%d_%H-%M-%S", time.localtime())
 
 # Parse command line arguments
 parser = argparse.ArgumentParser()
 parser.add_argument("--port", type=str, default="/dev/ttyACM0")
 parser.add_argument("--baud_rate", type=int, default=115200)
+parser.add_argument("--log_dir", type=str, default="./logs")
+parser.add_argument("--csv_file_name", type=str, default=f"imu_data_{date_str}.csv")
 args = parser.parse_args()
 
+# Create log directory if it does not exist
+if not os.path.exists(args.log_dir):
+    os.makedirs(args.log_dir)
+
+# Full path to the CSV file
+file_path = os.path.join(args.log_dir, args.csv_file_name)
+
 # Open serial port
 try:
     ser = serial.Serial(args.port, args.baud_rate)
 except SerialException as ex:
     print(ex.strerror)
     exit()
 
-# Allow for Arduino to reset
+# Allow for Arduino to reset (physically press the reset button)
 time.sleep(3)
 ser.reset_input_buffer()
 
 # Read data from Arduino
 try:
-    print("Starting logger.")
-    while True:
-        if ser.in_waiting > 0:  # Check if there is data to read
-            line = (
-                ser.readline().decode("utf-8").rstrip()
-            )  # Read a line from the serial
-            print(line)  # Print the line to the console
+    print(f"Starting logger. Writing to {file_path}")
+    with open(file_path, "w", newline="") as csv_log_file:
+        writer = csv.writer(csv_log_file)
+        writer.writerow(
```

Comment:
One thing that I am not particularly keen about is assuming knowledge of the headers. EX: timestamp, accelx, accely. What if code in tx changes?